src/main/java/com/unb/tracker/validator/UserValidator.java

```diff
@@ -0,0 +1,40 @@
+package com.unb.tracker.validator;
+
+import com.unb.tracker.model.User;
+import com.unb.tracker.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+
+@Component
+public class UserValidator implements Validator {
+	@Autowired
+	private UserService userService;
+
+	@Override
+	public boolean supports(Class<?> aClass) {
+		return User.class.equals(aClass);
+	}
+
+	@Override
+	public void validate(Object o, Errors errors) {
+		User user = (User) o;
+
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "email", "Email field cannot be empty.");
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "username", "Username field cannot be empty.");
+		if (user.getUsername().length() < 6 || user.getUsername().length() > 32)
+			errors.rejectValue("username", "Username must be between 6 and 32 characters.");
+
+		if (userService.findByUsername(user.getUsername()) != null)
+			errors.rejectValue("username", "Someone already has that username.");
+
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "password", "Password field cannot be empty.");
+		if (user.getPassword().length() < 8 || user.getPassword().length() > 32)
+			errors.rejectValue("password", "Password must be at least 8 characters in length.");
```

Comment:
If we're going to put a max for password length we should probably indicate that in the error message. The alternative is to just truncate it

Reply:
We should never truncate a password. If we did that, the user wouldn't understand why they can't log in. I'll add the message for password max length :)